addons/stock/report/report_stock_forecasted.py

```diff
@@ -186,44 +186,49 @@ def _reconcile_out_with_ins(lines, out, ins, demand, only_matching_move_dest=Tru
 
         lines = []
         for product in (ins | outs).product_id:
+            product_rounding = product.uom_id.rounding
             for out in reserved_outs_per_product[product.id]:
-                current = currents[out.product_id.id]
+                # Reconcile with reserved stock.
+                current = currents[product.id]
                 reserved = out.product_uom._compute_quantity(out.reserved_availability, product.uom_id)
                 currents[product.id] -= reserved
                 lines.append(self._prepare_report_line(reserved, move_out=out, reservation=True))
 
             unreconciled_outs = []
             for out in outs_per_product[product.id]:
                 # Reconcile with the current stock.
-                current = currents[out.product_id.id]
                 reserved = 0.0
                 if out.state in ('partially_available', 'assigned'):
                     reserved = out.product_uom._compute_quantity(out.reserved_availability, product.uom_id)
                 demand = out.product_qty - reserved
+
+                if float_is_zero(demand, precision_rounding=product_rounding):
+                    continue
+                current = currents[product.id]
                 taken_from_stock = min(demand, current)
-                if not float_is_zero(taken_from_stock, precision_rounding=product.uom_id.rounding):
+                if not float_is_zero(taken_from_stock, precision_rounding=product_rounding):
                     currents[product.id] -= taken_from_stock
                     demand -= taken_from_stock
                     lines.append(self._prepare_report_line(taken_from_stock, move_out=out))
                 # Reconcile with the ins.
-                if not float_is_zero(demand, precision_rounding=product.uom_id.rounding):
-                    demand = _reconcile_out_with_ins(lines, out, ins_per_product[out.product_id.id], demand, only_matching_move_dest=True)
-                if not float_is_zero(demand, precision_rounding=product.uom_id.rounding):
+                if not float_is_zero(demand, precision_rounding=product_rounding):
+                    demand = _reconcile_out_with_ins(lines, out, ins_per_product[product.id], demand, product_rounding, only_matching_move_dest=True)
```

Comment:
You really need to pass product_rounding here since we always pick the product uom

Reply:
It is a small optimization to avoid thousand and thousand (lookup on uom_id + uom_id.rounding) of cache lookup (in this case it matters)